e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1241,6 +1241,64 @@ func ResourceGroupStatusEquals(expected v1alpha1.ResourceGroupStatus) Predicate
 	}
 }
 
+// ResourceGroupHasObjects checks whether the objects are in the ResourceGroup's inventory.
+func ResourceGroupHasObjects(objects []client.Object) Predicate {
+	return func(obj client.Object) error {
+		rg, ok := obj.(*v1alpha1.ResourceGroup)
+		if !ok {
+			return WrongTypeErr(obj, &v1alpha1.ResourceGroup{})
+		}
+		var unselected []core.ID
+		for _, o := range objects {
+			selected := false
+			for _, res := range rg.Spec.Resources {
+				if res.Namespace == o.GetNamespace() && res.Name == o.GetName() &&
+					res.Group == o.GetObjectKind().GroupVersionKind().Group &&
+					res.Kind == o.GetObjectKind().GroupVersionKind().Kind {
+					selected = true
+					break
+				}
+			}
+			if !selected {
+				unselected = append(unselected, core.IDOf(o))
+			}
+		}
+
+		if len(unselected) == 0 {
+			return nil
+		}
```

Comment:
The code will be cleaner if the for loop becomes a function that returns a `bool` (found or not), which can be shared by the function below as well.

Reply:
If we use a shared function, then we need to iterate through all items. 
Currently, we can break the loop when found or not found without iterating through all items.

Reply:
The function can use a set/map to and then 1 loop to check whether the item can be found. This would avoid the need of `unselected` and make the code look cleaner.

Reply:
converted to use map.

The `unselected` is used to capture all objects missing from the ResourceGroup spec, which may not be necessary. After the change, it returns error as long as one object is missing.